File: Baek/impossible/Baek2096.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let lineLeft = null;
let prevMinScore = [0, 0, 0];
let prevMaxScore = [0, 0, 0];

rl.on('line', function(line) {
  if (lineLeft === null) {
    lineLeft = +line;
    return;
  }
  if (lineLeft === 0) return;

  const scores = line.split(" ").map(Number);

  const [pa1, pa2, pa3] = prevMinScore;
  const [pb1, pb2, pb3] = prevMaxScore;

  const curMinScore = scores.slice();
  const curMaxScore = scores.slice();
  curMinScore[0] += Math.min(pa1, pa2);
  curMinScore[1] += Math.min(pa1, pa2, pa3);
  curMinScore[2] += Math.min(pa2, pa3);
  prevMinScore = curMinScore;
  curMaxScore[0] += Math.max(pb1, pb2);
  curMaxScore[1] += Math.max(pb1, pb2, pb3);
  curMaxScore[2] += Math.max(pb2, pb3);
  prevMaxScore = curMaxScore;

  lineLeft--;
}).on("close", function() {
  console.log(Math.max(...prevMaxScore), Math.min(...prevMinScore));
  process.exit();
});

File: Baek/solved/Baek2096.py
N = int(input())

minScore = [0, 0, 0]
maxScore = [0, 0, 0]
for _ in range(N):
  a, b, c = list(map(int, input().split()))
  minScore = [
    a + min(minScore[0], minScore[1]),
    b + min(minScore[0], minScore[1], minScore[2]),
    c + min(minScore[1], minScore[2]),
  ]
  maxScore = [
    a + max(maxScore[0], maxScore[1]),
    b + max(maxScore[0], maxScore[1], maxScore[2]),
    c + max(maxScore[1], maxScore[2]),
  ]

print(str(max(maxScore)) + " " + str(min(minScore)))
